static/app/components/events/contexts/knownContext/user.tsx

```diff
@@ -0,0 +1,121 @@
+import {getContextKeys} from 'sentry/components/events/contexts/utils';
+import {t} from 'sentry/locale';
+import type {KeyValueListData} from 'sentry/types/group';
+import {defined} from 'sentry/utils';
+
+enum UserContextKeys {
+  ID = 'id',
+  EMAIL = 'email',
+  USERNAME = 'username',
+  IP_ADDRESS = 'ip_address',
+  NAME = 'name',
+  GEO = 'geo',
+}
+
+export interface UserContext {
+  // Any custom keys users may set
+  [key: string]: any;
+  [UserContextKeys.ID]?: string;
+  [UserContextKeys.EMAIL]?: string;
+  [UserContextKeys.USERNAME]?: string;
+  [UserContextKeys.IP_ADDRESS]?: string;
+  [UserContextKeys.NAME]?: string;
+  [UserContextKeys.GEO]?: Partial<Record<UserContextGeoKeys, string>>;
+}
+
+enum UserContextGeoKeys {
+  CITY = 'city',
+  COUNTRY_CODE = 'country_code',
+  SUBDIVISION = 'subdivision',
+  REGION = 'region',
+}
```

Comment:
I'm going to add these to the serializer since these are used all over but not rendered on issue details for some reason.